test/cmocka/src/audio/mixer/mixer_test.c

```diff
@@ -271,8 +271,7 @@ static void test_audio_mixer_copy(void **state)
 			samples[smp] = ((sin(rad) + 1) / 2) * (0xFFFFFFFF / 2);
 		}
 
-		tc->sources[src_idx].buf->stream.avail =
-			tc->sources[src_idx].buf->stream.size;
+		audio_stream_produce(&tc->sources[src_idx].buf->stream, sizeof(uint32_t) * MIX_TEST_SAMPLES);
```

Comment:
This feels like an actual change. Care to explain? (Won't block on it since it's a test case).

Reply:
@paulstelian97 Previously, whole buffer was mocked up and stitched together, just to make it work. I've modified the test, to make use of and actual comp_buffer, created via official API. Then audio_stream is properly configured and initiated, which allows for regular use of produce, to update stream state, including free and avail values, without the need to do it by hand.

Reply:
Fair enough.